backend/src/Utils/Datatypes.hs

```diff
@@ -28,18 +30,103 @@ data Occupancy = Occupancy { oTitle :: String -- ^ The title of the event, inten
                            , oEnd   :: UTCTime -- ^ When does this occupancy end?
                            } deriving (Show)
 
--- | Represents a single timeslot of half an hour
-newtype TimeSlot = TimeSlot UTCTime deriving (Show)
+-- | Represents a single timeslot of arbitrary length
+data TimeSlot = TimeSlot { tStart :: UTCTime -- ^ Start timestamp of the @TimeSlot@
+                         , duration :: NominalDiffTime -- ^ Duration of the @TimeSlot@
+                         } deriving (Show, Eq, Ord)
 
 instance Eq Occupancy where
   -- All fields but title considered
+  (==) :: Occupancy -> Occupancy -> Bool
   a == b = oStart a == oStart b && oEnd a == oEnd b
 
 instance Ord Occupancy where
   -- All fields but title considered
+  (<=) :: Occupancy -> Occupancy -> Bool
   a <= b = oStart a < oStart b || oStart a == oStart b && oEnd a <= oEnd b
 
 
+-- | Represents datatypes which have a start and end time.
+-- Is accompanied with many helpful functions on those times.
+class (Ord a, Eq a) => TimeSlice a where
```

Comment:
It would be great to add a pragma to this class to denote which methods are required and which is provided by the class. You could also move the methods that have a definition outside of the class, like with `disj` and `conj`.
```haskell
{-# MINIMAL (start, end | range), setStart, setEnd #-}
```